queue_job/migrations/10.0.1.0.0/end-migration.py

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, SUPERUSER_ID
+
+
+def migrate(cr, version):
+    """Now that everything has been loaded, compute the value of
+    channel_method_name.
+    """
+    if not version:
+        return
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    jobs = env['queue.job'].search([])
```

Comment:
Please don't iterate over the full history of jobs. Even on my smallest customer using the connector I have over 250000 jobs. Can you compute the string for each model once and then assign in SQL?

Reply:
No, each one is unique depending on its model and arguments. If we want to have a proper value, it should be done this way. We can fill with a default value, but then new jobs will be different from old ones.

Reply:
@pedrobaeza Maybe using a read_group on model_name. This can drastically improve assignation, no ?

Reply:
The problem is that they are not ORM/SQL fields, but you can see at the code, Python attributes, got dynamically. I will try to improve this grouping by `model_name, method_name` anyway.